src/main.rs

```diff
@@ -0,0 +1,73 @@
+use std::io::{ Stdin, Stdout, Write };
+
+fn main() {
+    loop {
+        let mut terminal1 = Terminal::new();
+        let todo = terminal1.ask_for_new_todo();
+        terminal1.show_todo(&todo);
+    }
+}
+
+#[derive(Debug, Clone)]
+struct Todo {
+    message: String,
+}
+
+impl Todo {
+    fn new(message: String) -> Self {
+        Todo { message }
+    }
+}
+
+struct Terminal {
+    stdin: Stdin,
+    stdout: Stdout,
+}
+
+impl Terminal {
+    fn new() -> Self {
+        Terminal {
+            stdin: std::io::stdin(),
+            stdout: std::io::stdout(),
+        }
+    }
+
+    fn ask_for_new_todo(&mut self) -> Todo {
+
+        if self.should_ask_for_new_todo() == false {
```

Comment:
Você não precisa comparar booleanos assim. Use `!`.
```suggestion
        if !self.should_ask_for_new_todo() {
```